merkledag/ipld.md

```diff
@@ -26,12 +26,20 @@ Objects with merkle-links form a Graph (merkle-graph), which necessarily is both
 
 ### What is a _merkle-path_?
 
-A _merkle-path_ is a unix-style path (e.g. `/a/b/c/d`) which initially dereferences through a _merkle-link_ and then follows _named merkle-links_ in the intermediate objects. Following a name means looking into the object, finding the _name_ and resolving the associated _merkle-link_.
+A merkle-path is a unix-style path (e.g. `/a/b/c/d`) which initially dereferences through a _merkle-link_ and allows access of elements of the referenced node and other nodes transitively.
+
+Merkle paths aren't suited to be used in filesystem representations (fuse mounts, HTTP or FTP protocols) as they describe the underlying IPLD data structure. Their use in filesystems is howver well suited for debug purposes (like `/proc` on unix).
```

Comment:
when we say this, do we mean:
- these paths are not good to represent _files_ because they allow access to the raw structures underneath
- or, these paths are not good to be accessed via the web or the filesystems, because there are UI problems.

i do want to be able to use these paths via HTTP to inspect the underlying data structures. but it's ok if i cannot get a proper file or directory representation out of it.

the latter part of this statement makes me somewhat more at peace with any of:
- disallowing _access of_ link properties via the paths
- disallowing transparent dereferencing without the use of `$obj/link/`
- separating "the link" and "the link properties" (same as the previous one in a way)

---

i think what we really need here is more concrete datastructure examples, and see how the pathing would want us to get to the stuff. maybe this will show us which solution is the best and guide our thinking.

perhaps we could write:
- an fs example (directories, sharded files, file attributes in dir entries)
- a version control example (commit histories)
- a social network example (users, messages, relationships) (maybe something like a simple version of foaf)
- a crypto network example (keys, derived keys, signatures, certificates, attestations) (i'll write this one at least)
- a blockchain example (block histories, transactions, etc)